zeppelin-interpreter/src/test/java/org/apache/zeppelin/interpreter/RemoteZeppelinServerResourceTest.java

```diff
@@ -0,0 +1,150 @@
+        assertEquals("1234", deserialized.getOwnerKey());
+        assertEquals(mapOf("key1", "value", "key2", "value2"), deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withNothing() {
+        // When only the empty JSON object is provided,
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson("{}");
+
+        // Then, all fields should be null
+        assertNull(deserialized.getOwnerKey());
+        assertNull(deserialized.getResourceType());
+        assertNull(deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withNullFields() {
+        String jsonWithNullFields = a2q("{" +
+                "'ownerKey':null," +
+                "'resourceType':null," +
+                "'data':null" +
+                "}");
+
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(jsonWithNullFields);
+        assertNull(deserialized.getOwnerKey());
+        assertNull(deserialized.getResourceType());
+        assertNull(deserialized.getData());
+    }
+
+    @Test
+    void toJson_withEmptyData() {
+        RemoteZeppelinServerResource original = new RemoteZeppelinServerResource();
+        original.setOwnerKey("");
+        original.setData(new TreeMap<>());
+        original.setResourceType(PARAGRAPH_RUNNERS);
+
+        String serializedJSON = original.toJson();
+        assertEquals(a2q("{" +
+                "'ownerKey':''," +
+                "'resourceType':'PARAGRAPH_RUNNERS'," +
+                "'data':{}" +
+                "}"), serializedJSON);
+
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(serializedJSON);
+        assertEquals(original.getOwnerKey(), deserialized.getOwnerKey());
+        assertEquals(original.getResourceType(), deserialized.getResourceType());
+        assertEquals(original.getData(), deserialized.getData());
+    }
+
+    @Test
+    void fromJson_withLargeDataSet() {
+        Map<String, String> largeData = new TreeMap<>();
+        for (int i = 0; i < 1000; i++) {
+            largeData.put("key" + i, "value" + i);
+        }
+
+        RemoteZeppelinServerResource original = new RemoteZeppelinServerResource();
+        original.setOwnerKey(UUID.randomUUID().toString());
+        original.setData(largeData);
+        original.setResourceType(PARAGRAPH_RUNNERS);
+
+        String serializedJSON = original.toJson();
+        RemoteZeppelinServerResource deserialized = RemoteZeppelinServerResource.fromJson(serializedJSON);
+
+        assertEquals(original.getOwnerKey(), deserialized.getOwnerKey());
+        assertEquals(original.getResourceType(), deserialized.getResourceType());
+        assertEquals(original.getData(), deserialized.getData());
+    }
+
+    private Map<String, String> mapOf(
+            String key, String value,
+            String key2, String value2
+    ) {
+        // TreeMap to ensure serialization order
```

Comment:
The ClusterMessageSerDeserTest uses a HashMap. Are there no issues like this

Reply:
Hmmmm🤔 , I don't think so. 
Or do you have anything in mind?